torch_xla/csrc/convolution.cpp

```diff
@@ -42,41 +45,90 @@ tensorflow::ConvOpAttrs MakeConvOpAttrs(
   return conv_op_attrs;
 }
 
-std::vector<xla::int64> FilterTransposePermutation() { return {2, 3, 1, 0}; }
+// Transpose filter shape to have [channel, batch] as last two dimensions.
+// 4D case: (N, C, H, W) -> (H, W, C, N)
+const std::vector<xla::int64>& FilterTransposePermutation(const xla::int64 k) {
+  XLA_CHECK(k == 4 || k == 5) << k;
```

Comment:
Sorry for late change ... I think this (and below) reads a bit better as:
```C++
if (k == 4) {
  ...
} else if (k == 5) {
  ...
} else {
  XLA_ERROR() << "Invalid rank: " << k;
}
```

IOW, drop the `XLA_CHECK()` and folds into an `XLA_ERROR()`.
I can change it later if you don't want to do it here.